hudi-common/src/main/java/org/apache/hudi/common/table/log/AbstractHoodieLogRecordReader.java

```diff
@@ -241,7 +241,12 @@ private void scanInternalV1(Option<KeySpec> keySpecOpt) {
     try {
       // Iterate over the paths
       logFormatReaderWrapper = new HoodieLogFormatReader(fs,
-          logFilePaths.stream().map(logFile -> new HoodieLogFile(new CachingPath(logFile))).collect(Collectors.toList()),
+          logFilePaths.stream()
```

Comment:
Should the same filtering logic be wrapped into a util method given it is also used in `scanInternalV2`?

Reply:
Fix it by checking both the file name and the block instant.

Reply:
Where are you comparing with block instant time? 
If i understand correctly, `logFile.getDeltaCommitTime` will return the base instant time for older log files, while it will return deltacommit time for newer versions right? 
If the caller of log record reader creates file slices based on completion time and instantiates the log record reader with only committed `logFilePaths`, then there should be no need to check here.

Reply:
We do not have that general completion filtering for current file slice APIs, not sure if it is safe to do that, I did see some APIs to expose the inflight files.

Reply:
Fixed by introducing the APIs.